implementations/java-openapiprocessor/build.gradle.kts

```diff
@@ -12,8 +14,15 @@ repositories {
     }
 }
 
-kotlin {
-    jvmToolchain(libs.versions.build.jdk.get().toInt())
+tasks.withType<KotlinCompile> {
+    kotlinOptions {
+        jvmTarget = "11"
```

Comment:
Honestly I don't know why they decided to set target for 11 and not 17. I guess it shouldn't matter at all it's just the bytecode version for harness and not for their lib. I think generally any code compiled with older jdk version should work without any issues on newer jdk versions. I could just make it to compile do jdk 17 bytecode, but I don't know :D

Reply:
I would assume (again not knowing anything and possibly not being arsed to check -- which means feel free to ignore as well!) that compiling for older JVMs probably means forgoing e.g. optimizations which may exist at the bytecode layer but only work on newer ones.

I.e. I'd expect compiling for older JVMs increases compatibility and potentially decreases performance. And that here since we're running in a single fixed JVM version that compiling for the newest one possible is the smartest.

But yeah I know nothing, I'm just guessing -- it likely doesn't matter. If it works when you change this to 17 I'd personally go with that, but if it doesn't seem to work for even trivial reasons then yeah leave it and ignore.

Reply:
Yeah, probably your reasoning about performance is correct. At least it would be for plain java, I'm not sure if this also the same for kotlin. Nevertheless I agree that compiling to newest bytecode would be best. Gonna try it out later :)